game-03/components/platform/float_platform.tscn

```diff
@@ -48,47 +47,131 @@ _data = {
-children = Array[ExtResource("3_r18qn")]([SubResource("Resource_qhpan")])
+children = Array[ExtResource("3_r18qn")]([SubResource("Resource_71dy5"), SubResource("Resource_r4ifk")])
 arguments = {
 "method_name": &"play_shake"
 }
 
-[sub_resource type="Resource" id="Resource_0pxlj"]
+[sub_resource type="Resource" id="Resource_bvj0i"]
 script = ExtResource("3_6j5ma")
-root = SubResource("Resource_31up2")
+root = SubResource("Resource_15x7q")
 canvas_position = Vector2(100, 75)
 
 [sub_resource type="Resource" id="Resource_6q7jc"]
 script = ExtResource("5_a0j4o")
 script_inherits = "AnimationPlayer"
-block_serialization_trees = Array[ExtResource("3_6j5ma")]([SubResource("Resource_0pxlj")])
+block_serialization_trees = Array[ExtResource("3_6j5ma")]([SubResource("Resource_bvj0i")])
 variables = Array[ExtResource("6_yyx0f")]([])
 generated_script = "extends AnimationPlayer
 
 
 func play_shake():
-	if \"ahead\" == \"ahead\":
+	if 'forward' == \"forward\":
 		play('shake')
 	else:
 		play_backwards('shake')
+	if 'until done' == \"until done\":
+		await animation_finished
+
+	(get_node(\"../..\")).queue_free()
+
+"
+version = 0
+
+[sub_resource type="RectangleShape2D" id="RectangleShape2D_3ufhb"]
+size = Vector2(256, 8)
+
+[sub_resource type="Resource" id="Resource_qh7d5"]
+script = ExtResource("5_83svu")
+name = &"area2d_on_entered:something"
+arguments = {}
+
+[sub_resource type="Resource" id="Resource_e6pu7"]
+script = ExtResource("5_83svu")
+name = &"is_node_in_group"
+arguments = {
+"group": "player",
+"node": SubResource("Resource_qh7d5")
+}
+
+[sub_resource type="Resource" id="Resource_xq1nr"]
+script = ExtResource("5_83svu")
+name = &"get_node"
+arguments = {
+"path": NodePath("../RigidBody2D/AnimationPlayer")
+}
+
+[sub_resource type="Resource" id="Resource_5a1us"]
+script = ExtResource("3_r18qn")
+name = &"call_method_node"
+children = Array[ExtResource("3_r18qn")]([])
+arguments = {
+"method_name": "play_shake",
+"node": SubResource("Resource_xq1nr")
+}
+
+[sub_resource type="Resource" id="Resource_d8dep"]
+script = ExtResource("3_r18qn")
+name = &"if"
+children = Array[ExtResource("3_r18qn")]([SubResource("Resource_5a1us")])
+arguments = {
+"condition": SubResource("Resource_e6pu7")
+}
+
+[sub_resource type="Resource" id="Resource_dm1fd"]
+script = ExtResource("3_r18qn")
+name = &"area2d_on_entered"
+children = Array[ExtResource("3_r18qn")]([SubResource("Resource_d8dep")])
+arguments = {}
+
+[sub_resource type="Resource" id="Resource_jegfr"]
+script = ExtResource("3_6j5ma")
+root = SubResource("Resource_dm1fd")
+canvas_position = Vector2(75, 75)
+
+[sub_resource type="Resource" id="Resource_cri4e"]
+script = ExtResource("5_a0j4o")
+script_inherits = "Area2D"
+block_serialization_trees = Array[ExtResource("3_6j5ma")]([SubResource("Resource_jegfr")])
+variables = Array[ExtResource("6_yyx0f")]([])
+generated_script = "extends Area2D
+
+
+func _init():
+	body_entered.connect(_on_body_entered)
 
+func _on_body_entered(something: Node2D):
```

Comment:
No particular need to fix this but the animation triggers when the player first enters the box, even if that's from below. I would expect it to trigger only when the player actually stands on the platform. I think that would be hard to do currently.

Reply:
Indeed.  It is hard to get the collision direction right now.